karapace/schema_reader.py

```diff
@@ -242,6 +244,9 @@ def run(self) -> None:
                     self.offset = self._get_beginning_offset()
                 try:
                     self.handle_messages()
+                except ShutdownException:
+                    self._stop_schema_reader.set()
+                    shutdown()
                 except Exception as e:  # pylint: disable=broad-except
```

Comment:
Adding the shutdown exception raises the question that what would be the action to take on the fallback `Exception`?

Reply:
Yes totally agree 👍 , normally in scenarios like this, we should fire a metric to track the exceptions and then form an alert around the various exceptions. We can perform another iteration, ideally, this work will trigger the review of our instrumentation.